### PR TITLE
Fix: vertical bar broken in chart/gear acitivity (issue: #1646)

### PR DESCRIPTION
**Pull request by AdityaKrSingh26**
Fixed the broken vertical line in Chart/Gears activities, ensuring its applicable on all plaforms (issue: #1646)

Steps:
- Divided the toolbar in two parts left and right, used` dislay: flex` and `justify-content: space-between` to move both the part to ends of both side of screen. and used `dislay: flex` and `align-item:centre` to move hr to correct position.
- Earlier the negative margin was causing the cross platform issue for hr. Removed the negative margin.

### Chrome:
image
image

### Firefox:
image
image

### Electron.js app:
image
image

### CODE REVIEW

File: activities/Chart.activity/index.html [hunk 1]
```diff
@@ -21,45 +21,49 @@
...
+					<sugar-toolitem
+						ref="sharedPal"
+						id="network-button"
+						:title="l10n.stringNetwork"
+						palette-file="sugar-web/graphics/presencepalette"
+						palette-class="PresencePalette"
+						palette-event="shared"
+						@shared="SugarPresence.onShared"
+						v-if="SugarPresence"
+					></sugar-toolitem>
```
Comment (llaske) [inline activities/Chart.activity/index.html:35]: Do not change indentation.

File: activities/Chart.activity/index.html [hunk 2]
```diff
@@ -21,45 +21,49 @@
...
-					v-bind:title="l10n.stringExportSettings" 
-					palette-file="lib/js/palettes/exportpalette.js"
-					palette-class="ExportPalette"
-					palette-event="export-file"
-					@export-file="exportFile"
-					:title="l10n.stringExportSettings"
-				></sugar-toolitem>
+					<hr />
+					<sugar-toolitem id="add-button" :title="l10n.stringAddValue" @click="addData"></sugar-toolitem>
+					<sugar-toolitem id="remove-button" :title="l10n.stringRemoveValue" @click="removeData"></sugar-toolitem>
```
Comment (llaske) [inline activities/Chart.activity/index.html:39]: Do not change indentation.

File: activities/Chart.activity/lib/sugar-web/graphics/css/sugar-96dpi.css [hunk 3]
```diff
@@ -26,16 +26,23 @@ a {
...
+  display: flex;
+  align-items: center;
+}
 /* Toolbar separator */
 .toolbar hr {
   display: inline-block;
   height: 33px;
   margin: 2.5px;
-  margin-bottom: -17px;
+  /* margin-bottom: -17px; */
```
Comment (llaske) [inline activities/Chart.activity/lib/sugar-web/graphics/css/sugar-96dpi.css:45]: Do not let commented code.

**Review by llaske — CHANGES_REQUESTED** [review]
I guess it's related to your IDE settings but do not change file indentation.
Only lines updated should be in the PR.

**Comment by AdityaKrSingh26** [comment]
Ok  i will make the changes you mentioned. Thanks

**Comment by AdityaKrSingh26** [comment]
Hey @llaske i have removed the comments and indentation in my latest commit please have a look

**Comment by llaske** [comment]
It works.
You're not supposed to change the library `sugar-96dpi.css` because it's a shared library copied in every activity.
BTW it could be acceptable if it's a minor change. But using `flex` in the toolbar a major change that will impact all activities. Plus split it in left/right has no logical reason. 
Could you find a way to limit the impact to activity css only or to the `hr` element only?

**Comment by AdityaKrSingh26** [comment]
I looked around a bit and found out that the issue with <hr/ > lines not being consistent was `margin-bottom : -17px` the negative margin is causing cross platform issue as different browser interpret negative margin differently (please correct me if i am wrong). While` display : flex `property is consistent on all platforms.

I split right/left because on applying display flex the style class of pull-right was over written so to move the icons in group in 2 sides of screen i used `justify-content : space-between`

**Comment by AdityaKrSingh26** [comment]
One thing that i can think of is that wrapping <hr/> tag with a <div> tag and apply flex property on parent <div> of <hr/>
Is this fine ?

**Comment by llaske** [comment]
I understand your fix but I would like a fix where only the CSS `.toolbar hr` need to be updated.
Upgrading HTML and use a `flex` property will change too many files.

**Comment by AdityaKrSingh26** [comment]
Hey @llaske .... i looked and tried out diffrent css properties for restricting the changes to `<hr/>` tag only but it seems pretty impossible without changing some properties of the parent tag.

As you mentioned we are not supposed to modify the `sugar-96dpi.css` one thing we can do is add the necessary changes in `acitivity.css` of the specific required files

**Comment by llaske** [comment]
@AdityaKrSingh26 that's fine if you can limit change to `activity.css`

**Comment by AdityaKrSingh26** [comment]
@llaske please have a look at the latest commit.

**Comment by llaske** [comment]
It works but I'm not a big fan of using flex because it change the look and the behavior of the toolbar.
Why not replace the HR object by a DIV with a vertical border? It will avoid changing the toolbar itself.

**Comment by AdityaKrSingh26** [comment]
> It works but I'm not a big fan of using flex because it change the look and the behavior of the toolbar. Why not replace the HR object by a DIV with a vertical border? It will avoid changing the toolbar itself.

Ok i will make the changes

**Comment by AdityaKrSingh26** [comment]
> It works but I'm not a big fan of using flex because it change the look and the behavior of the toolbar. Why not replace the HR object by a DIV with a vertical border? It will avoid changing the toolbar itself.

Done. Please have a look

**Comment by AdityaKrSingh26** [comment]
@llaske  Done :+1: 
Firefox:
Screenshot from 2024-12-18 13-57-16


Chrome:
Screenshot from 2024-12-18 13-57-01


Electron:
Screenshot from 2024-12-18 13-56-34

**Comment by llaske** [comment]
Work also on Safari.
Good job, thanks.

**Comment by AdityaKrSingh26** [comment]
> Work also on Safari. Good job, thanks.

Finally. :sob:

**Pull request merged**